src/bms_sender.py

```diff
@@ -2,6 +2,7 @@
 import random
 import os
 import json
+import re
 
 class BMS_Sender:
```

Comment:
which one function/method to call so that sender sends data to console?

Reply:
def create_temp_stream_output(self, idx, battery_idx, bat) function, would print the output in a stream
Please check the link : https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/blob/master/src/bms_sender.py
for the function

Reply:
Hi @sunilgowda95 , I have taken your points into consideration and modified the script, now the function printing to console is **def create_stream_output**

Reply:
have moved the printing function to a separate : **def print_console_output**, for adding exceptional handling
and modified the test case
FYI @sunilgowda95 @sudeeprp

Reply:
Hi,
Can you share me a function that I can run to get 50 values simulated data stream from sender.
This function is needed to achieve sender and receiver integration.

Following lines in ReadMe.md gives the requirement:
"The Sender sends data to the Receiver using [console redirection and pipes](https://ss64.com/nt/syntax-redirection.html). They need to run on the command-line as follows:

sender-executable | receiver-executable"

Console redirection can be achieved using following command for python:
 "python3 <sender>.py | python3 bms_receiver.py"

Currently I am running test/bms_sender.test.py for checking data from sender in console for the receiver.

Reply:
Hi @sunilgowda95 , can you place the screenshot/ link on where the piped output is shown?

Reply:
Hi @omprakashs855 I am not able to pipe the data from sender without running the sender test scripts.
sender-executable | receiver-executable --> from [README.md](https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855#readme).

We should not use/rely on test scripts for integration.

Expecting a quicker reply.

Reply:
> Hi @sunilgowda95 , can you place the screenshot/ link on where the piped output is shown?

Sender and receiver integration can be Piped using github actions file. That I have not created as I do not see one function/executable (other than the test code) which will generate data for the receiver.

Reply:
Hi @sunilgowda95 , Please check the below repo : https://github.com/clean-code-craft-tcq-7/stream-line-UdhayPT
In this I have not received any Sender Output ( and as the code is in Java, I have created my own sender console output from Receiver), from here you can see how I have used the pipe operation

The pipe is performing one operation, taking output from console and providing it as input to receiver, do ping me if you need any help with the logic

Reply:
Hi @omprakashs855 I would like to iterate the requirements given in README.md file. It Clearly says sender has to simulate and send 50 values of at least 2 battery parameters.

https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/blob/master/src/bms_sender.py --> this file only has sender class. I do not see one method which will simulate data for my receiver (yes it has all the methods needed but not one function which will create data for receiver).

Kindly create one function in the file src/bms_sender.py just like how you have used your sender class to initialize and create data in test/bms_sender.test.py's test_bms_sender_console_regex.

https://github.com/clean-code-craft-tcq-7/stream-line-sunilgowda95
you can check above repo where I have created sender and  Srinath Sridhar  has used it to create his receiver and successfully integrated using console redirection. He has not written any code for sender to achieve integration.

If you have not got sender code in the assigned repo please check with the developer.

Also for showcasing the integration I have used sender test and created the integration workflow here : https://github.com/clean-code-craft-tcq-7/stream-line-omprakashs855/actions/runs/4340658063/jobs/7579408959.
Please check the output.

Once I get a sender method which sends data to receiver I can replace "test/bms_sender.test.py" with it.

Reply:
Hi @sunilgowda95, could you start with the receiver output script, to be created independent of the sender console output,
I wanted to see how you want the input data, once I can see how you need the output, we can start on the integration

Do ping me on my NTID : IOM2KOR (if you need any more input)

P.S : we need to stream the input data, and have streaming output data one by one

Reply:
Hi @omprakashs855  Integration of sender and receiver is done. Thank you for all the updates.